packages/core/src/auth/auth.ts

```diff
@@ -282,6 +282,8 @@ export class Auth implements AuthService, ConnectionManager {
         }
 
         const id = randomUUID()
+        void this.updateConnectionState(id, 'authenticating')
```

Comment:
'authenticating' is also the state in `authenticate()`:
https://github.com/aws/aws-toolkit-vscode/blob/0bbfa1320b7a073ad688f246058fe171c5a51be4/packages/core/src/auth/auth.ts#L723

are the semantics the same. is it correct to always set 'authenticating'  whenever `createConnection()` is called?


also directly below we're passing 'unauthenticated' ...

Reply:
> is it correct to always set 'authenticating' whenever createConnection() is called?
tbh, I think this is the right approach as we should store intermediate state of `unauthenticated`.

The `unauthenticated` state is stored in `tokenProvider` though is not updated in state.

Looking at the function `updateConnectionState` --
https://github.com/aws/aws-toolkit-vscode/blob/75d0d159c8597559763cb97a61a9e8e36768336c/packages/core/src/auth/auth.ts#L507C5-L512C10

was storing the `unauthenticated` value because `this.store.getProfileOrThrow(id)` throw error and have an early return.

Reply:
We should avoid touching `auth.ts` unless required. Some more info on why this was needed will help.

Since the connection does not exist yet, I'm wondering if it makes sense to update the state. No one knows about this new connection at this point.

Reply:
The idea is to store the intermediate step so that we can display a notification if they attempt to do an auth and still close the chat window.